packages/ubuntu_wizard/lib/src/utils/string.dart

```diff
@@ -0,0 +1,22 @@
+/// Utility extensions for [String].
+extension StringOrIfEmpty on String {
+  /// Returns the string if it's not empty, otherwise returns the [other] string.
+  String orIfEmpty(String other) => isEmpty ? other : this;
+
+  /// Sanitizes a string for use in user- and hostnames.
+  String sanitize() {
+    // Replace all non-alphanumeric characters with hyphens.
+    var sanitized = replaceAll(RegExp('[^a-zA-Z0-9-]'), '-');
+    // Remove leading hyphens.
+    while (sanitized.startsWith('-')) {
```

Comment:
notice that it's a while loop. all leading hyphens are removed

Reply:
I missed the while loop :p sorry about that. But if you run the regexp to remove all 2+ consecutive hypens first, you don’t need it, correct?

Reply:
i added a test for "  myname  "

Reply:
:+1: but you don’t want to remove the while loops I guess?

Reply:
hmm, with the test cases we have so far, it doesn't make any difference. i wanted to leave as short string as possible for the regexp processing, but i don't mind flipping it around if it's easier to read that way?

Reply:
> I missed the while loop :p sorry about that. But if you run the regexp to remove all 2+ consecutive hypens first, you don’t need it, correct?

and I missed this comment haha. removing leading and trailing hyphens is still needed because after removing all occurrences of "--", there might still be a singular leading or trailing hyphen.

Reply:
I don’t have a strong opinon, both rationale makes sense (having the regexp being lighter vs removing the loop) and it was a nitpick :)

Well obviously, I missed the while reading it as an "if" first, but that’s probably only me. Thanks for additional test making it clear it works! :)